relay-ourlogs/src/ourlog.rs

```diff
@@ -35,7 +35,7 @@ pub fn otel_to_sentry_log(otel_log: OtelLog) -> Result<OurLog, Error> {
         time_unix_nano,
         ..
     } = otel_log;
-    let span_id = SpanId(hex::encode(span_id));
+    let span_id = SpanId::try_from_bytes(&span_id)?;
```

Comment:
I'm returning early here on parsing failure, same as for the `trace_id`. Would it make more sense to use `Annotated` here to catch the error case and not return early?